lib/puppet/pops/types/types.rb

```diff
@@ -364,6 +402,16 @@ def eql?(o)
 # @api public
 #
 class PType < PTypeWithContainedType
+
+  def self.register_ptype(loader, ir)
+    create_ptype(loader, ir, 'AnyType',
+       'type' => {
+         KEY_TYPE => POptionalType.new(PType::DEFAULT),
+         KEY_VALUE => nil
+       }
+    )
+  end
```

Comment:
It is derived from PTypeWithContainedType - cannot all derived classes from PTypeWithContainedType use one implementation?
There were several occurrences of self.register_ptype with what looked like the very same call in several classes derived from PTypeWithContainedType.